### PR TITLE
Ef 249 kvsplugin experiment

### PR DESCRIPTION
**Pull request by chenhudl**
- Pepijn's audio hack for plugin
- Remove kvs stream function from the plugin (non-webrtc related)
- Rename kvsplugin to kvswebrtcplugin
- Add droppable log
- Add support for key control
- Fix log level

### CODE REVIEW

File: gst/gst-kvs-plugin/src/GstPlugin.c [hunk 1]
```diff
@@ -228,11 +206,6 @@ VOID gst_kvs_plugin_class_init(GstKvsPluginClass* klass)
                                                         DEFAULT_REGION,
                                                         (GParamFlags)(G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS)));
 
-    g_object_class_install_property(gobject_class, PROP_LOG_LEVEL,
-                                    g_param_spec_uint("log-level", "Logging Level", "Logging Verbosity Level", LOG_LEVEL_VERBOSE,
-                                                      LOG_LEVEL_SILENT + 1, LOG_LEVEL_SILENT + 1,
-                                                      (GParamFlags)(G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS)));
-
```
Comment (pandreidoru) [inline gst/gst-kvs-plugin/src/GstPlugin.c:314]: Are we sure that we'll not use this logging?

File: gst/gst-kvs-plugin/src/GstPlugin.c [hunk 2]
```diff
@@ -700,18 +659,18 @@ GstFlowReturn gst_kvs_plugin_handle_buffer(GstCollectPads* pads, GstCollectData*
...
+        pGstKvsPlugin->firstPts = buf->pts;
+    }
 
-        buf->pts += pGstKvsPlugin->producerStartTime - pGstKvsPlugin->firstPts;
+    if (pGstKvsPlugin->startTime == GST_CLOCK_TIME_NONE) {
+        pGstKvsPlugin->startTime = GETTIME() * DEFAULT_TIME_UNIT_IN_NANOS;
     }
 
+    buf->pts += pGstKvsPlugin->startTime - pGstKvsPlugin->firstPts;
+    buf->pts -= pGstKvsPlugin->firstPts;
```
Comment (pandreidoru) [inline gst/gst-kvs-plugin/src/GstPlugin.c:671]: Is this extra line a mistake?

File: gst/gst-kvs-plugin/src/GstPlugin.h [hunk 3]
```diff
@@ -157,7 +148,6 @@ struct __GstParams {
     gchar* accessKey;
     gchar* secretKey;
     gchar* awsRegion;
-    guint rotationPeriodInSeconds;
     guint logLevel;
```
Comment (pandreidoru) [inline gst/gst-kvs-plugin/src/GstPlugin.h:151]: Shouldn't `logLevel` be removed also?

Comment (pandreidoru) [reply gst/gst-kvs-plugin/src/GstPlugin.h:151]: I'm not sure about `fileLogPath` ...

File: gst/gst-kvs-plugin/src/GstPlugin.h [hunk 4]
```diff
@@ -324,15 +259,13 @@ struct __GstKvsPlugin {
     UINT64 lastDts;
     UINT64 basePts;
```
Comment (pandreidoru) [inline gst/gst-kvs-plugin/src/GstPlugin.h:260]: We don't need this one.

Comment (pandreidoru) [inline gst/gst-kvs-plugin/src/GstPlugin.h:260]: We don't need this one.

File: gst/gst-kvs-plugin/src/GstPlugin.c (hunk 1 above)
Comment (chenhudl) [reply gst/gst-kvs-plugin/src/GstPlugin.c]: Initially setting log-level had no effect, so I removed it. I add it back now and also set the environment variable properly.

File: gst/gst-kvs-plugin/src/GstPlugin.h (hunk 3 above)
Comment (chenhudl) [reply gst/gst-kvs-plugin/src/GstPlugin.h:152]: see above

File: gst/gst-kvs-plugin/src/GstPlugin.c (hunk 2 above)
Comment (chenhudl) [reply gst/gst-kvs-plugin/src/GstPlugin.c]: I think so. I don't remember why it end up there, dont see it in original file either

Comment (pandreidoru) [reply gst/gst-kvs-plugin/src/GstPlugin.c]: I tried to make sense of that math ... and checked the original file also ...

**Review by pandreidoru — APPROVED** [review]

**Pull request merged**